### ISSUE TITLE
Toggling slash notation on drum part doesn't turn the selected notes into slash notation

### ISSUE THREAD

**Issue by runarcn** [musescore/MuseScore#21502]
### Issue type

Engraving bug (incorrect score rendering)

### Bug description

When writing something into the drum part and toggling slash notation, the notes only get put on the center line instead of that + having their noteheads turned into slashes. If it's written into vo3 it gets put at the top, and for vo4 at the bottom. Afterwards it's possible to manually change their noteheads, though I have had them turn back into what I originally changed them from after working on the score for a while.

### Steps to reproduce

1. Write or copy over some notes into the drums
2. Hit "toggle rhythmic slash notation"
3. Watch

### Screenshots/Screen recordings

Adding into voice 3 and clicking "Tools > Toggle rhythmic slash notation"
![image](https://github.com/musescore/MuseScore/assets/52424029/fc7804aa-b454-4b06-860e-c91c4d3cde4c)

Copying over a voice from one of the horns and clicking "Tools > Toggle rhythmic slash notation"
![image](https://github.com/musescore/MuseScore/assets/52424029/9ea231c1-c28f-4641-ac98-96276fc2f59b)

### MuseScore Version

OS: Fedora Linux 39 (Workstation Edition), Arch.: x86_64, MuseScore version (64-bit): 4.2.1-240230938, revision: github-musescore-musescore-d757433

### Regression

I don't know

### Operating system

Fedora Linux 39 (Workstation Edition) x86_64 

### Additional context
[copy I made when reporting this bug.zip](https://github.com/musescore/MuseScore/files/14269890/copy.I.made.when.reporting.this.bug.zip)

[score i'm working on.zip](https://github.com/musescore/MuseScore/files/14269934/score.i.m.working.on.zip) (see replies)

**Comment by MarcSabatella** [comment]
Please ZIP and attach the specific score you are having trouble with.  I cannot reproduce this, so I am guessing it's something unique to your score and its drumset definition.  Also tell us exactly which notes you are entering and selecting

Note that it important to select only the voice you wish to convert.  The selection filter can be useful for this if you are trying to select only voice 3 when you already have content in voice 1.

**Comment by runarcn** [comment]
Zip files:
[score i'm working on.zip](https://github.com/musescore/MuseScore/files/14269934/score.i.m.working.on.zip)

[copy I made when reporting this bug.zip](https://github.com/musescore/MuseScore/files/14269890/copy.I.made.when.reporting.this.bug.zip)


The recording shows ride cymbal hits that I've selected and clicked the "Toggle rhythmic slash notation button" on, but not gotten slash note heads for. I've then manually changed the noteheads to the Slash notehead before it reverts itself later on.

However I wonder if that is bugged for this score in general and not just the drumset? I tried doing the exact same for Electric Guitar and Acoustic Bass just to be sure that it worked elsewhere, and got this result (Measure 131 - 132):

https://github.com/musescore/MuseScore/assets/52424029/c712bda0-b977-471d-914b-31b09a1954fc

**Comment by NicolaRulli** [comment]
A currently know bug is:
1. If parts have been created, all new "toggle rhythmic slash notation" commands will cause noteheads to be the opposite of what the toggle demands (centering and play marking on the other hand are correct)
2. In that bugged scenario, some notes in the drumset staff will occasionally revert to non-slash noteheads after having been correctly set, possibly only when the part is open, possibly due to non-rhythmic-slash-notation drumset staff noteheads being by default resistant to notehead changes in the properties panel. In that case, you should notice that, although you gave no command to the notes in the part, they're highlighted in bright orange as if desynced.

**Comment by runarcn** [comment]
> A currently know bug is:
> 
>     1. If parts have been created, all new "toggle rhythmic slash notation" commands will cause noteheads to be the opposite of what the toggle demands (centering and play marking on the other hand are correct)
> 
>     2. In that bugged scenario, some notes in the drumset staff will occasionally revert to non-slash noteheads after having been correctly set, possibly only when the part is open, possibly due to non-rhythmic-slash-notation drumset staff noteheads being by default resistant to notehead changes in the properties panel. In that case, you should notice that, although you gave no command to the notes in the part, they're highlighted in bright orange as if desynced.

This sounds like it! In my case however, the drum part was not opened while number 2. happened. I also can't recall anything about notes being in bright orange as if desycned though.

Is there any known workaround to this as of right now?

**Comment by MarcSabatella** [comment]
Confirmed.  The issue with parts is already fixed for the next update but drumset does add a new wrinkle because the workaround of manually changing the noteheads isn't supported.

**Comment by NicolaRulli** [comment]
> Confirmed. The issue with parts is already fixed for the next update but drumset does add a new wrinkle because the workaround of manually changing the noteheads isn't supported.

It may not be supported, but for some reason it partially works. Trying over and over seems to fix it eventually. One needs a few passes, though.

**Comment by NicolaRulli** [comment]
> can't recall anything about notes being in bright orange as if desycned though.

Maybe I'm getting confused with another parts bug, but I'm pretty sure I observed it.

**Comment by MarcSabatella** [comment]
A better workaround may be, add a new drumset staff, *don't* generate a part for it, enter your notation there and toggle the rhythmic slash notation - this will work correctly, more or less - then copy/paste.

I say it "more or less" works because stem directions won't necessarily be correct.  I found entering a mix of quarter note and eighth notes as snare then toggling yields slash notation with the quarter notes stems down, the eighths stem up:

![image](https://github.com/musescore/MuseScore/assets/1799936/9a219da4-d633-4cd4-b9dd-4e60c3691198)

**Comment by runarcn** [comment]
> A better workaround may be, add a new drumset staff, _don't_ generate a part for it, enter your notation there and toggle the rhythmic slash notation - this will work correctly, more or less - then copy/paste.

Great thanks, I'll do that once I get to the "make sure the engraving is good and stuff" part before sending it out.

Should I close the issue?

**Comment by MarcSabatella** [comment]
I would say, not until someone verifies correct behavior in a nightly/master build.  My guess is all will be well except maybe the stem directions, which would be worth updating the original description to clarify if so.

I'll check it out and report back shortly...

**Comment by MarcSabatella** [comment]
The basic issue with the noteheads is indeed fixed in master, but not the stem direction issue.  That has nothing to do with parts, though - same issue in a score with no parts generated at all.  And, that's not a recent regression - same in 4.1.1.  But it was better in MU3.

You could either close this and then we can open a new issue for the stems, or you could edit the title and description to reflect the stem issue only.

**Comment by oktophonie** [comment]
A new issue should be opened. This one is already far too noisy.
